### PR TITLE
Adding stage to path, fixes #4 'Try it out'

### LINKED ISSUE #4: API execution not working ('try it out')
When I execute any API, it does not take the stage context into consideration , hence it fails . 

Eg. - 
Swagger - http://localhost:3000/dev-b/swagger
API : http://localhost:3000/dev-b/test 

When I execute it from swagger (try it out) it runs - http://localhost:3000/test 

Do I need to add any extra configuration(base path) to fix this ?

### PR DESCRIPTION
**Pull request by NicoIamarino**
I just added stage to url that will be used to 'Try it out'

### CODE REVIEW

**Comment by NicoIamarino** [comment]
#4

**Review by bfaulk96 — APPROVED** [review]
This would be really nice – can't really use this package without it

**Comment by bfaulk96** [comment]
Also, I don't think this fixes #4 – they specifically need a basePath, not a stage, from the sounds of it.

**Comment by NicoIamarino** [comment]
> Also, I don't think this fixes #4 – they specifically need a basePath, not a stage, from the sounds of it.

The URL generated by API Gateway includes the stage so to 'try it out' you moust use the same route

File: src/ServerlessAutoSwagger.ts [hunk 1]
```diff
@@ -249,6 +249,10 @@ class ServerlessAutoSwagger {
         const functions = this.serverless.service.functions;
         Object.entries(functions).map(([functionName, config]) => {
             const events = config.events || [];
+            
+            let stagePath = this.serverless.configurationInput?.provider?.stage
+            stagePath = stagePath ? `/${stagePath}` : ""
```
Comment (SamWSoftware) [inline src/ServerlessAutoSwagger.ts:254]: I think we can do this in a nicer way with a 'basePath' defined in the 'server' settings which are global to the whole API. 
https://swagger.io/specification/#server-object

**Review by SamWSoftware — CHANGES_REQUESTED** [review]
i think we can use base path in the server config to do this globally instead of in ever function

**Comment by bfaulk96** [comment]
@NicoIamarino @SamWSoftware https://github.com/SamWSoftware/serverless-auto-swagger/pull/32 should fix this in a slightly nicer way 😄

**Comment by bfaulk96** [comment]
Closing this, as Swagger's `basePath` has been implemented in 1.7.0 (which will be published soon hopefully)

**Pull request closed without merging**